File: ReactApp/projekat/src/services/ItemService.js
import axios from "axios";
import { config } from "./UserService";

export const AddItemBackend = async (item) =>
{
    return await axios.post(process.env.REACT_APP_API_URL + '/api/items/create', item, config);
}

export const GetItemsBySellerId = async (sellerId) =>
{   
    return await axios.get(process.env.REACT_APP_API_URL + '/api/items/' + sellerId, config);
}

export const DeleteItem = async (id) =>
{   
    return await axios.delete(process.env.REACT_APP_API_URL + '/api/items/' + id, config);
}

export const UpdateItem = async (id, item) =>
{
    return await axios.put(process.env.REACT_APP_API_URL + '/api/items/' + id, item, config);
}

export const GetAllItems = async () =>
{   
    return await axios.get(process.env.REACT_APP_API_URL + '/api/items/all', config);
}

export const GetItemsByOrderId = async (orderId) =>
{   
    return await axios.get(process.env.REACT_APP_API_URL + '/api/items/byOrder/' + orderId, config);
}

File: Projekat/Projekat/Services/ItemService.cs
using AutoMapper;
using Microsoft.EntityFrameworkCore;
using Projekat.Data;
using Projekat.Dto;
using Projekat.Interfaces;
using Projekat.Models;
using Projekat.Repository;

namespace Projekat.Services
{
    public class ItemService : IItemService
    {

        private readonly IMapper _mapper;
        private readonly ItemRepo _itemRepo;

        public ItemService(IMapper mapper, ItemRepo itemRepo)
        {
            _mapper = mapper;
            _itemRepo = itemRepo;
        }

        public ItemDto CreateItem(ItemDto itemCreate)
        {
            try
            {
                Item item = _mapper.Map<Item>(itemCreate);
                _itemRepo.Add(item);
                _itemRepo.Save();

                return _mapper.Map<ItemDto>(item);
            }
            catch (Exception)
            {
                return null;
            }

        }

        public List<ItemDto> GetItemsBySellerId(long sellerId)
        {
            try
            {
                return _mapper.Map<List<ItemDto>>(_itemRepo.GetAll(sellerId));
            }
            catch (Exception)
            {
                return null;
            }
        }

        public bool DeleteItem(long id)
        {
            try
            {
                Item item = _itemRepo.Get(id);

                _itemRepo.Delete(item);

                _itemRepo.Save();

                return true;
            }
            catch (Exception)
            {
                return false;
            }

        }

        public ItemDto UpdateItem(long id, ItemDto newItem)
        {
            try
            {
                Item noviItem = _mapper.Map<Item>(newItem);

                Item itemDB = _itemRepo.Update(id, noviItem);

                _itemRepo.Save();

                return _mapper.Map<ItemDto>(itemDB);
            }
            catch (Exception)
            {
                return null;
            }
        }

        public ItemDto UpdateItemAfterOrder(long id, int amount)
        {
            Item itemDB = _itemRepo.Get(id);
            itemDB.Amount -= amount;
            _itemRepo.Save();

            return _mapper.Map<ItemDto>(itemDB);
        }


        public List<ItemDto> GetAll()
        {
            try
            {
                return _mapper.Map<List<ItemDto>>(_itemRepo.GetAllItems());
            }
            catch (Exception)
            {
                return null;
            }
        }

        public ItemDto GetItemById(long id)
        {
            return _mapper.Map<ItemDto>(_itemRepo.Get(id));
        }

        public List<ItemDto> GetItemsByOrderId(long orderId)
        {
            try
            {
                List<ItemsInsideOrderDto> itemsInsideOrderDto = _mapper.Map<List<ItemsInsideOrderDto>>(_itemRepo.GetAllItemsInsideOrder(orderId));
                List<ItemDto> items = new List<ItemDto>();
                foreach (var item in itemsInsideOrderDto)
                {
                    ItemDto itemDB = GetItemById(item.ItemId);
                    itemDB.Amount = item.Amount;
                    items.Add(itemDB);
                }
                return items;
            }
            catch (Exception)
            {
                return null;
            }

        }
    }
}
